src/puppetlabs/jdbc_util/core.clj

```diff
@@ -48,6 +48,67 @@
      (future-cancel f#)
      result#))
 
+(defn user-exists?
+  "Given a DB spec that connects with a user besides `username`, return
+  a boolean indicating whether `username` exists."
+  [admin-db-spec username]
+  (-> (jdbc/query admin-db-spec ["SELECT 1 AS present FROM pg_roles WHERE rolname = ?" username])
+    first
+    :present
+    (= 1)))
+
+(defn create-user!
+  "Given a DB spec that has a user with permission to create roles and isn't
+  `username`, create the user `username` with the given `password`. Accepts an
+  optional options map where the :superuser? option will make the created user
+  a superuser if the value is exactly `true` (not just truthy); note that only
+  superusers can create more superusers."
+  ([admin-db-spec username password] (create-user! admin-db-spec username password {}))
+  ([admin-db-spec username password options]
+   (let [{:keys [superuser?]} options
+         sql (format (str "CREATE ROLE %s WITH LOGIN PASSWORD %s" (if (true? superuser?)
+                                                                    " SUPERUSER"))
+                     (pg-escape-identifier username)
+                     (pg-escape-string password))]
+     (jdbc/execute! admin-db-spec [sql])
+     nil)))
+
+(defn drop-user!
+  "Given a DB spec that connects with a user besides `username` and has
+  permisson to drop that user, remove the named user. Note that only superusers
+  can remove other superusers; other users can be removed by anyone with the
+  CREATEROLE permission."
+  [admin-db-spec username]
+  (let [sql (format "DROP ROLE IF EXISTS %s" (pg-escape-identifier username))]
+    (jdbc/execute! admin-db-spec [sql])
+    nil))
+
+(defn user-for-role-selection
+  "Given a user and role name, return an sql & params vec with a selection that
+  will return one row with the column `has_role` and the value `true` if the
+  user is a member of the role, and no rows otherwise."
+  [user role]
+  [(str "SELECT true AS has_role"
+        " WHERE EXISTS ("
+        "   SELECT member, roleid FROM pg_auth_members"
+        "   WHERE member IN ("
+        "     SELECT oid FROM pg_roles WHERE rolname = ?"
+        "   ) AND roleid IN ("
+        "     SELECT oid FROM pg_roles WHERE rolname = ?"
+        "   )"
+        " );"
+        )
+   user, role])
+
+(defn has-role?
+  "Returns true if the user is a member of the role, and false if not."
+  [db-spec user role]
```

Comment:
I think this misses a bunch of cases, like implicit membership (eg that a superuser has to all roles) or inheritance of more than one level.

You should be able to replace the whole function with
```sql
select pg_has_role(user, role, 'MEMBER')
```